src/Commands/CommandRepair.cs

```diff
@@ -24,59 +24,80 @@
 using System.Collections.Generic;
 using Essentials.Api.Command;
 using Essentials.Api.Command.Source;
-using Essentials.Api.Unturned;
-using Essentials.Common.Util;
 using SDG.Unturned;
-using Essentials.I18n;
-using Essentials.Common;
-using System.Reflection;
-
-namespace Essentials.Commands {
+using Rocket.API;
+using Rocket.Unturned.Player;
 
+namespace Essentials.Commands
+{
     [CommandInfo(
         Name = "repair",
         Aliases = new[] { "fix" },
         Description = "Repair all items in your inventory.",
-        AllowedSource = AllowedSource.PLAYER
+        AllowedSource = AllowedSource.BOTH
     )]
-    public class CommandRepair : EssCommand {
-
-        private readonly FieldInfo _itemsField = ReflectUtil.GetField<Items>("items");
-
-        public override CommandResult OnExecute(ICommandSource src, ICommandArgs args) {
-            var player = src.ToPlayer();
+    public class CommandRepair : IRocketCommand
+    {
+        public AllowedCaller AllowedCaller => AllowedCaller.Both;
+        public string Name => "repair";
+        public string Help => "Repair all items in your inventory.";
+        public string Syntax => "/repair";
+        public List<string> Aliases => new List<string>() { "fix" };
+        public List<string> Permissions => new List<string>() { "repair", "fix" };
 
-            player.Inventory.items.ForEach(item => Repair(player, item));
-            EssLang.Send(src, "ALL_REPAIRED");
+        public void Execute(IRocketPlayer caller, string[] command)
+        {
+            switch (command.Length)
+            {
+                case (0):
+                    if (caller is ConsolePlayer)
+                        Rocket.Core.Logging.Logger.LogException(new System.InvalidOperationException($"Console does not have an iventory to repair! You must be a player to execute /repair or /fix"));
```

Comment:
you typed iventory

Reply:
damn I thought you found something essential.